mtwitter.js

```diff
@@ -0,0 +1,254 @@
+  };
+  Request.post(options, function(error, response, body) {
+      if (error) {
+          console.log("Error", error);
+          return false;
+      } else {
+          try {
+              var jsonresponse = JSON.parse(body);
+              if (jsonresponse.token_type == "bearer") {
+                  fs.writeFile("bearer.tok", jsonresponse.access_token, (err) => {
+                      if (err) {
+                          console.error("Couldn't write bearer token to file. Reason:",err);
+                      }
+                  });
+                  cb(jsonresponse.bearer_token);
+              } else {
+                  console.error("Error getting bearer token: Unexpected response");
+                  cb(null);
+              }
+          } catch (e) {
+              console.error("Error getting bearer token:", e);
+              cb(null);
+          }
+
+      }
+  });
+}
+
+mtwitter.prototype.get_user_by_id = function(id) {
+    var ts = new Date().getTime();
+    return new Promise((resolve, reject) => {
+        for (var name in this.tuser_cache) {
+            var cached = this.tuser_cache[name];
+            if (ts - cached.cache_time > TWITTER_PROFILE_INTERVAL) {
+                continue;
+            }
+            if (cached.user != null && cached.user.id_str == id) {
+                resolve(cached.user);
+            }
+        }
+        this.app_twitter.get('users/show', {
+            user_id: id
+        }, (error, user, response) => {
+            if (error) {
+                console.error(error);
+                reject(error);
+            } else {
+                this.tuser_cache[user.screen_name] = {
+                    cache_time: ts,
+                    user: user
+                };
+                resolve(user);
+            }
+        });
+    });
+}
+
+mtwitter.prototype.get_user = function(name) {
+    console.log("Looking up @" + name);
+    return new Promise((resolve, reject) => {
+        var ts = new Date().getTime();
+        if (this.tuser_cache[name] != undefined) {
+            console.log("Checking cache for @" + name);
+            var cached = this.tuser_cache[name];
+            if (ts - cached.cache_time < TWITTER_PROFILE_INTERVAL) {
+                resolve(cached.user);
+            }
+        }
+
+        console.log("Checking twitter for @" + name);
+        this.app_twitter.get('users/show', {
+            screen_name: name
+        }, (error, user, response) => {
+            if (error) {
+                this.tuser_cache[name] = {
+                    cache_time: ts,
+                    user: null
+                };
+                reject(error);
+            }
+            this.tuser_cache[name] = {
+                cache_time: ts,
+                user: user
+            };
+            resolve(user);
+        });
+    });
+
+}
+
+/*
+  Add a user's timeline to the iterator. Tweets will be automatically send to
```

Comment:
> Add a user's timeline to the iterator

Er, what? I see two lists (no iterators) here. The function name is `enqueue` which makes me think that something is going to `dequeue` it, is that correct? If yes, then state what will happen when it is `dequeue`d. If no, then change the name of this function to be something like `add_timeline`.